### DISCUSSION TITLE
Strip path seperator at the end of result

### DISCUSSION THREAD

**Discussion by moritzdietz** [sharkdp/fd#1187] [Q&A] [answered]
With https://github.com/sharkdp/fd/releases/tag/v8.4.0 there was a change that directories have a trailing slash now.

This broke something in my setup and now I am looking for a trivial way to revert to the old behavior.
Has anyone an idea? Hoping there is a trivial way other than running the output through sed and the like.

**Comment by tmccombs** [comment, marked as answer]
Running the output through sed seems pretty trivial to me.

**Reply by moritzdietz** [reply]
Sure. It is easy to just pipe it to sed, but I wanted to know whether there is a command line flag that reverts to the old behavior as the linked issues on the change and related issues have lots of discussion about exactly this.

**Reply by tmccombs** [reply]
It doesn't look like there is currently. The final state of the PR does not have such an option.

I do think it would be possible to construct the path without the trailing /  with exec or exec-batch.

**Reply by moritzdietz** [reply]
Ok, thanks. I guess I will have to play around and do some testing. This was a kind of breaking change to me so I'm a bit annoyed but this is upstreams "fault" I guess :P

**Reply by tmccombs** [reply]
It does look like `fd -X echo '{}'` doesn't include the trailing "/". I'm not actually sure if that is intentional or not. 

You particular use case was not really considered when this change was made. It _might_ be a compelling reason to add a flag for this. But there are workarounds, so it isn't immediately clear that the value of being able to do this outweighs the cost of adding yet another flag.

For reference https://github.com/sharkdp/fd/pull/812#issuecomment-968317568 has the reasoning for why the flag wasn't included.

**Reply by moritzdietz** [reply]
To give a full example of what my usecase is exactly.

I am using `fd` in combination with `parallel` from the GNU core utilities [moreutils](https://packages.debian.org/sid/moreutils) package.
As the `parallel` man page states `parallel - build and execute shell command lines from standard input in parallel` I feed it a list of Git repository directories and have `parallel` run a Git command in batches of 30 at the same time.

This used to look like this:
```
fd -I -a --type d --hidden '^\.git$' /home/moritz/Documents/git | parallel -j 30 --progress 'echo -ne "\t{.} - " && git --work-tree={.} --git-dir={} pull'
```
Now, after the upstream dependency changed the behavior, I need `sed` in between my two commands to remove the trailing `/` at the end of each path to a directory:
```
fd -I -a --type d --hidden '^\.git$' /home/moritz/Documents/git | sed 's/\/$//' | parallel -j 30 --progress 'echo -ne "\t{.} - " && git --work-tree={.} --git-dir={} pull'
```
Here is a screenshot of the individual steps and the generated output. At the end is the final command which I usually run. Imagine this list having hundreds of repositories…

![image](https://user-images.githubusercontent.com/7935536/210338316-655714b7-c91a-42eb-b9a7-e1e26046584f.png)

I can live with the `sed` solution even though I don't like it. But then again, as you already stated: `fd` **is** opinionated and I guess I don't agree with its opinion :P

**Comment by tmccombs** [comment]
FWIW, if https://github.com/sharkdp/fd/pull/1043 was merged, then you could probably just use `--format {.}` to get what you want. 

p.s. I think that you are actually using [GNU parallel](https://www.gnu.org/software/parallel/), not moreutils parallel.  the moreutils parallel doesn't have a `--progress` option.
